src/Common/src/TypeSystem/Interop/IL/MarshalUtils.cs

```diff
@@ -59,6 +59,10 @@ public static bool IsBlittableType(TypeDesc type)
         public static bool IsManagedSequentialType(TypeDesc type)
```

Comment:
This is becoming more and more specific to R2R/CoreCLR layout rules. Would be better to move this to the R2R specific file?

Reply:
For now I have removed the superfluous new condition in IsBlittableType. For IsManagedSequentialType, that has always been only used by the R2R compiler. Are you thinking about something along the lines of CorInfoImpl.ReadyToRun / RyuJit or would you prever moving the IsManagedSequentialType check to a completely new class?

Reply:
I would move it to `ReadyToRunMetadataFieldLayoutAlgorithm.cs`. It is specific to the current CoreCLR field layout algorithm.

Reply:
I am temporarily rolling back all changes to MarshalUtils as they require more work and delay merging in other useful parts of this change, most notably fixes for import table consistency and for the large version bubble. I'm going to send this change out in a separate future PR after performing the comparative analysis with Crossgen par Jan's suggestion.